spring-cloud-starter-stream-processor-object-detection/src/main/java/org/springframework/cloud/stream/app/object/detection/processor/ObjectDetectionTensorflowOutputConverter.java

```diff
@@ -0,0 +1,174 @@
+
+	private final String[] labels;
+	private float confidence;
+	private List<String> modelFetch;
+
+	public ObjectDetectionTensorflowOutputConverter(Resource labelsResource, float confidence, List<String> modelFetch) {
+		this.confidence = confidence;
+		this.modelFetch = modelFetch;
+		try {
+			this.labels = loadLabels(labelsResource);
+			Assert.notNull(this.labels, String.format("Failed to initialize object labels [%s].", labelsResource));
+		}
+		catch (Exception e) {
+			throw new RuntimeException(String.format("Failed to initialize object labels [%s].", labelsResource), e);
+		}
+
+		logger.info(String.format("Object labels [%s] loaded.", labelsResource));
+	}
+
+	/**
+	 * Loads object labels in the string_int_label_map.proto
+	 * @param labelsResource
+	 * @return
+	 * @throws Exception
+	 */
+	private static String[] loadLabels(Resource labelsResource) throws Exception {
+		try (InputStream is = labelsResource.getInputStream()) {
+			String text = StreamUtils.copyToString(is, Charset.forName("UTF-8"));
+			StringIntLabelMapOuterClass.StringIntLabelMap.Builder builder =
+					StringIntLabelMapOuterClass.StringIntLabelMap.newBuilder();
+			TextFormat.merge(text, builder);
+			StringIntLabelMapOuterClass.StringIntLabelMap proto = builder.build();
+
+			int maxLabelId = proto.getItemList().stream()
+					.map(StringIntLabelMapOuterClass.StringIntLabelMapItem::getId)
+					.max(Comparator.comparing(i -> i))
+					.orElse(-1);
+
+			String[] labelIdToNameMap = new String[maxLabelId + 1];
+			for (StringIntLabelMapOuterClass.StringIntLabelMapItem item : proto.getItemList()) {
+				if (!StringUtils.isEmpty(item.getDisplayName())) {
+					labelIdToNameMap[item.getId()] = item.getDisplayName();
+				}
+				else {
+					// Common practice is to set the name to a MID or Synsets Id. Synset is a set of synonyms that
+					// share a common meaning: https://en.wikipedia.org/wiki/WordNet
+					labelIdToNameMap[item.getId()] = item.getName();
+				}
+			}
+			return labelIdToNameMap;
+		}
+	}
+
+	@Override
+	public Tuple convert(Map<String, Tensor<?>> tensorMap, Map<String, Object> processorContext) {
+
+		try (Tensor<Float> scoresTensor = tensorMap.get(DETECTION_SCORES).expect(Float.class);
+			 Tensor<Float> classesTensor = tensorMap.get(DETECTION_CLASSES).expect(Float.class);
+			 Tensor<Float> boxesTensor = tensorMap.get(DETECTION_BOXES).expect(Float.class);
+		) {
+			// All these tensors have:
+			// - 1 as the first dimension
+			// - maxObjects as the second dimension
+			// While boxesT will have 4 as the third dimension (2 sets of (x, y) coordinates).
+			// This can be verified by looking at scoresT.shape() etc.
+			int maxObjects = (int) scoresTensor.shape()[1];
+			float[] scores = scoresTensor.copyTo(new float[1][maxObjects])[0];
+			float[] classes = classesTensor.copyTo(new float[1][maxObjects])[0];
+			float[][] boxes = boxesTensor.copyTo(new float[1][maxObjects][4])[0];
+
+			// Work in progress
```

Comment:
Should we expect something else here to be done?

Thanks

Reply:
This is to enable support for image segmentation feature: https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/instance_segmentation.md
While the TF models already support this and we can read the metadata, it turned to be quite involving to convert the mask metadata into mask images.
So this fragment can be regarded as an enabler to support the image segmentation in some of the following up releases. 
I will improve the  descriptions.